docs/content/topics/scripting/guide.md

```diff
@@ -9,1401 +9,867 @@ aliases:
-})
+});
 
-events.on("push", () => {
+events.on("brigade.sh/github", "push", async event => {
   console.log(" **** I'm a GitHub 'push' handler")
-})
+});
 ```
-[brigade-03.js](../../examples/brigade-03.js)
 
 Now if we re-run our `brig` command, we will see the same output as before:
 
 ```
-Started brigade-worker-01brx5m1yppb0dxn4emk76jqtv-master
-yarn start v0.27.5
-$ node prestart.js
-prestart: src/brigade.js written
-$ node --no-deprecation ./dist/src/index.js
-Creating PVC named brigade-worker-01brx5m1yppb0dxn4emk76jqtv-master
-==> handling an 'exec' event
-Destroying PVC named brigade-worker-01brx5m1yppb0dxn4emk76jqtv-master
-Done in 1.21s.
-```
-
-Since Brigade did not see a `push` event, it did not execute the `push` event handler.
-It only executed the `exec` handler that `brig` causes.
-
-### Where Do Events Come From?
-
-In order to be able to write good Brigade scripts, we need to know what events we
-can listen for. So what is the origin of these events?
-
-To answer this question, we need a tiny bit of background knowledge about Brigade. Brigade has
-several components in its runtime. The _worker_ runs the JavaScript that we are writing
-here. The _controller_ starts and manages the _worker_. But there are also Brigade
-_gateways_.
+$ brig event create --project first-event --follow
 
-A Brigade gateway is a piece of code that executes in your cluster and generates events.  An
-overview of a few Brigade gateways ready for use can be seen in the [Gateways doc](./gateways.md).
+Created event "6769b7fd-ddd0-4a42-aa17-723c02a2b61a".
 
-The `brig` client declares its own hook (`exec`).
-
-You can create your own gateways which generate their own events, or use gateways created
-by others.
-
-In the list above, there are two special events that have very specific usage: `after` and
-`error`.
-
-### Two Special Events: _after_ and _error_
-
-The `after` and `error` events are what are called _lifecycle hooks_. They let you
-execute some code when Brigade hits a certain stage of processing.
-
-The `after` hook runs _at the very end_ of any session that did not die from an error.
-
-```javascript
-const { events } = require("brigadier")
-
-events.on("exec", () => {
-  console.log("==> handling an 'exec' event")
-})
-
-events.on("after", () => {
-  console.log(" **** AFTER EVENT called")
-})
+Waiting for event's worker to be RUNNING...
+2021-09-20T22:29:28.498Z INFO: brigade-worker version: a398ba8-dirty
+Hello, World!
 ```
-[brigade-04.js](../../examples/brigade-04.js)
 
-The `brig` client will trigger the `exec` event. But then when that event has been
-processed, Brigade will trigger an `after` event before returning:
+Since Brigade did not see a GitHub push event (in this case, an event from
+source `brigade.sh/github` and of type `push`), it did not execute the
+corresponding event handler. It only executed the handler for the event created
+by `brig`, i.e. an event from source `brigade.sh/cli` and of type `exec`.
```

Comment:
"Did not see" is another case where I worry we could be misrepresenting the event/gateway model. Could we reword this section to something like the following:

> Since the event we created was from `brigade.sh/cli` and of type `exec`, only the corresponding handler was executed. The handler for events from `brigade.sh/github` and of type `push` was not.